src/vulkan/graphics_pipeline.cc

```diff
@@ -571,8 +591,24 @@ Result GraphicsPipeline::Draw(const DrawArraysCommand* command,
   if (instance_count == 0 && command->GetVertexCount() != 0)
     instance_count = 1;
 
-  vkCmdDraw(command_->GetCommandBuffer(), command->GetVertexCount(),
-            instance_count, command->GetFirstVertexIndex(), 0);
+  if (command->IsIndexed()) {
+    if (!index_buffer_)
+      return Result("Vulkan: Draw indexed is used without given indices");
+
+    r = index_buffer_->BindToCommandBuffer(command_->GetCommandBuffer());
+    if (!r.IsSuccess())
+      return r;
+
+    vkCmdDrawIndexed(command_->GetCommandBuffer(),
+                     command->GetVertexCount(),      /* indexCount */
+                     instance_count,                 /* instanceCount */
+                     0,                              /* firstIndex */
```

Comment:
Should this be GetFirstVertexIndex()?

Reply:
No. I was also very confused because of this. VkRunner spec says
```
vertexCount will be used as the index count, firstVertex becomes the vertex offset and firstIndex will always be zero.
```

To explain this, I added a test case as tests/cases/draw_triangle_list_with_index_buffer_and_vertex_offset.amber.
I will add more comments to explain it clearly.